go/new_merkletree/merkle_tree_test.go

```diff
@@ -0,0 +1,228 @@
+package new_merkletree
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"hash"
+	"log"
+	"reflect"
+	"testing"
+)
+
+/* DummyDAO is a NewMerkleTree DAO type which simply returns an uppercase character
+ * for each leaf in the tree.  For obvious reasons, this makes it a little tricky
+ * to support trees larger than 26 leaves, but for testing purposes that shouldn't
+ * be a terrible burden.
+ */
+type DummyDAO struct {
+	NewMerkleTreeDataInterface
+
+	vals []byte
+}
+
+func NewDummyDAO(size int) *DummyDAO {
+	dao := new(DummyDAO)
+	dao.vals = []byte("ABCDEFGHIJKLMNOPQRSTUVWXYZ")[0:size]
+	return dao
+}
+
+func (dao DummyDAO) EntryAt(i uint64) ([]byte, error) {
+	return dao.vals[i : i+1], nil
+}
+
+func (dao DummyDAO) Size() uint64 {
+	return uint64(len(dao.vals))
+}
+
+/* NullHash is a stupidly simple "hash" algorithm that simply returns its input
+ * as the output.  Why is this useful?  Testing.
+ */
+type NullHash struct {
+	hash.Hash
+
+	s []byte
+}
+
+func (h *NullHash) Write(p []byte) (int, error) {
+	h.s = bytes.Join([][]byte{h.s, p}, []byte{})
+	return len(p), nil
+}
+
+func (h *NullHash) Sum(b []byte) []byte {
+	return bytes.Join([][]byte{h.s, b}, []byte{})
+}
+
+func (h *NullHash) Reset() {
+	h.s = []byte{}
+}
+
+func (h *NullHash) Size() int {
+	return len(h.s)
+}
+
+func (h *NullHash) BlockSize() int {
+	return 1
+}
+
+// Real testing methods start here.
+
+func TestNew(t *testing.T) {
+	var tree *NewMerkleTree
+	tree = New(nil, nil, sha256.New())
```

Comment:
I've removed one extraneous line, and added a comment about why the var is there.